### PR TITLE
Developer NVIDIA (Improving on Failure Reason and logic of Force Update and some testcases)

### PR DESCRIPTION
**Pull request by dhruvr-nvidia**
Changes:

  1. Multipart Force Update will happen only for backup or old version fwpkg.
  2. Added try exception block in precheck and verify functions.
  3. Improving on the Failure Reason.
  4. Improved Logic on F8 testcase so that it does not enter infinite loop.
  5. Minor bugfix in F88 testcase.

Tests:
Sanity/Regression tests were run.

### CODE REVIEW

File: ctam/interfaces/fw_update_ifc.py [hunk 1]
```diff
@@ -444,7 +454,7 @@ def RedFishFWUpdate(self, BinPath, URI, targets=[], is_multipart=False):
             headers = {"Content-Type": "multipart/form-data"}
             body = {
                 "UpdateFile": (BinPath, open(BinPath, "rb"), "application/octet-stream"),
-                "UpdateParameters" : ("Targets", json.dumps({"Targets": targets, "ForceUpdate": True if self.dut().multipart_force_update else False}),'application/json')
+                "UpdateParameters" : ("Targets", json.dumps({"Targets": targets, "ForceUpdate": True if is_force_update else False}),'application/json')
```
Comment (MeritedHobbit) [inline ctam/interfaces/fw_update_ifc.py:457]: The Redfish Firmware Update Whitepaper calls out Targets as a standard parameter. But is ForceUpdate a standard parameter?

Comment (MeritedHobbit) [inline ctam/interfaces/fw_update_ifc.py:457]: Are we saying that downgrade requires a ForceUpdate? Then we should this ask this as a standard?

Comment (dhruvr-nvidia) [reply ctam/interfaces/fw_update_ifc.py:457]: Yes for Multipart Update we need ForceUpdate for downgrade. We have the ForceUpdate under UpdateParameters in the redfish schema: https://redfish.dmtf.org/schemas/v1/UpdateService.v1_14_1.json
For update using HttpPushUri, ForceUpdate is not needed.

**Comment by MeritedHobbit** [comment]
We are almost done reviewing this PR. Apologies for the delay.

**Pull request merged**